inc/api.class.php

```diff
@@ -1308,6 +1337,42 @@ protected function getItems($itemtype, $params = [], &$totalcount = 0) {
       return array_values($found);
    }
 
+   /**
+    * Get raw names
+    *
+    * @param array  $data           A raw from the database
+    * @param array  $params         API parameters
+    * @param string $self_itemtype  Itemtype the API was called on
+    *
+    * @return array
+    */
+   protected function getFriendlyNames(
+      array $data,
+      array $params,
+      string $self_itemtype
+   ) {
+      $_names = [];
+
+      foreach ($params[self::PARAM_FORCENAME] as $fn_fkey) {
+         if ($fn_fkey == "self") {
+            // Get friendlyname for current item
+            $fn_itemtype = $self_itemtype;
+            $fn_id = $data["id"];
+         } else {
+            // Get friendlyname for given fkey
+            $fn_itemtype = getItemtypeForForeignKeyField($fn_fkey);
```

Comment:
You should validate that this is a valid itemtype and classname, for example by using `getItemForItemtype`.